### PR TITLE
Fix/error message when fail registration

### PR DESCRIPTION
**Pull request by jiachen1120**
@NicholasAzar @stevehu Hi, this PR is related to the error message when consul registration. I just let the server down by throwing an exception when consul registration fails so there would be no matter whether multiple ports are listening (what Frank reported). Look forward to your review! Thanks

The RFC related to this:
https://github.com/networknt/light-rfcs/blob/Fix/error-message-when-registration-failure/light-4j/0009-Fix-error-message-when-fails-to-register-in-consul.md

### CODE REVIEW

File: server/src/main/java/com/networknt/server/Server.java [hunk 1]
```diff
@@ -256,35 +263,36 @@ static private boolean bind(HttpHandler handler, int port) {
...
-                if (logger.isInfoEnabled())
-                    logger.info("Https Server started on ip:" + config.getIp() + " Port:" + port);
-            } else {
-                System.out.println("Https port disabled.");
+                // handle the registration exception separately to eliminate confusion
+            } catch (Exception e) {
+                System.out.println("Failed to register on Consul, server stopped.");
                 if (logger.isInfoEnabled())
-                    logger.info("Https port disabled.");
+                    logger.info("Failed to register on Consul, server stopped.");
```
Comment (NicholasAzar) [inline server/src/main/java/com/networknt/server/Server.java]: There shouldn't be any reference to Consul in here. The server isn't locked into a specific registration service

**Comment by NicholasAzar** [comment]
@jiachen1120 right now when the server can't bind to the configured port it won't exit

**Comment by NicholasAzar** [comment]
@jiachen1120 if consul is unavailable when the server is starting up, the server still starts up fine and binds to the port, however the following is continually output to the logs:
```
HOST IP null
12:06:57.409 [main] com.cibc.apif-hello-1   ERROR c.n.consul.client.ConsulClientImpl registerService - Exception:
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.xnio.nio.WorkerThread$ConnectHandle.handleReady(WorkerThread.java:321)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:561)
Http port disabled.
Https Server started on ip:0.0.0.0 Port:8443
12:06:59.069 [pool-3-thread-1]    ERROR c.n.consul.client.ConsulClientImpl checkPass - CheckPass request exception
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.xnio.nio.WorkerThread$ConnectHandle.handleReady(WorkerThread.java:321)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:561)
12:07:19.078 [pool-3-thread-2]    ERROR c.n.consul.client.ConsulClientImpl checkPass - CheckPass request exception
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.xnio.nio.WorkerThread$ConnectHandle.handleReady(WorkerThread.java:321)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:561)
12:07:39.078 [pool-3-thread-3]    ERROR c.n.consul.client.ConsulClientImpl checkPass - CheckPass request exception
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.xnio.nio.WorkerThread$ConnectHandle.handleReady(WorkerThread.java:321)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:561)
Disconnected from the target VM, address: '127.0.0.1:57933', transport: 'socket'
```
Also, the RFC linked in the original comment in this pr is 404ing

**Comment by jiachen1120** [comment]
Hey, the previous RFC has already been merged to
https://github.com/networknt/light-rfcs/blob/master/light-4j/0009-Fix-error-message-when-fails-to-register-in-consul.md

**Comment by stevehu** [comment]
Make a fix that reserve the previous behavior. When binding is failed, don't throw RuntimeException to stop the server but simply log one line message without the stack trace to indicate the binding to the port is failed, try the next port. Then return false to indicate that binding is failed so that the next port will be tried. In the second try/catch block for the consul registration, a RuntimeException is thrown and the server will be stopped. 

```
        } catch (Exception e) {
            System.out.println("Failed to bind to port " + port + ". Trying " + ++port);
            if (logger.isInfoEnabled())
                logger.info("Failed to bind to port " + port + ". Trying " + ++port);
            return false;
        }
```

**Pull request merged**